docs/feature-flags.md

```diff
@@ -0,0 +1,47 @@
+# Feature Flags
+
+Features inside the `wc-admin` repository can be in various states of completeness. In addition to the development copy of `wc-admin`, feature plugin versions are bundled, and code is merged to WooCommerce core. To provide a way for improved control over how these features are released in these different environments, `wc-admin` has a system for feature flags.
+
+We currently support the following environments:
+
+| Environment | Description                                                                                                                                                            |
+|-------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| development | Development - All features should be enabled in development. These flags are also used in both JS and PHP tests. Ran using `npm start`.                                |
+| plugin      | Plugin - A packaged release of the featured plugin, for GitHub WordPress.org. Ran using `npm run-script build:release`. |                                    |
+| core        | Core - assets/files ready and stable enough for core merge. @todo No build process exists yet.
+
+
+## Adding a new flag
+
+Flags can be added to the files located in the `config/` directory. Make sure to add a flag for each environment and explicitly set the flag to false.
+Please add new feature flags alphabetically so they are easy to find.
+
+## Basic Use - Client
+
+The `window.wcAdminFeatures` constant is a global variable containing the feature flags.
+
+Instances of `window.wcAdminFeatures` are replaced during the webpack build process by using webpack's [define plugin](https://webpack.js.org/plugins/define-plugin/). Using `webpack` for this allows us to eliminate dead code when making minified/production builds (`plugin`, or `core` environments).
+
+To check if a feature is enabled, you can simplify check the boolean value of a feature:
+
+```
+{ window.wcAdminFeatures[ 'activity-panels' ] && <ActivityPanel /> }
```

Comment:
I like the simplicity of just checking the global - but wondering if you considered adding a util method like `isFeatureEnabled` or something along those lines. Probably would result in more code throughout the app, but the only benefit I see there is being able to update the logic in one file vs dozens if/when we need to change it for some reason.

Reply:
@timmyc I definitely considered a utility method here but couldn't quite get it to work 100%. I just spent a bit more time confirming this.

The trade off here basically boils down to webpack's code elimination.

Right now, with this branch, if webpack encountered this line and the flag was turned off, this code would not end up in the plugin/production bundle. I tried a few different methods to try to get it to phase a function, but the code ends up being bundled (even if not displayed to the user).

One potential solution (also raised in the Gutenberg thread -- which I think they decided against) is to write a custom babel plugin that could transform these function calls into the global check before webpack hits it.

I'm personally fine just checking a variable. The extra work would basically be for different syntax since these flags should ever only be boolean values (so the function logic should never really change anyway). What do you think?

Reply:
✨thanks for the in-depth explanation - I hadn't considered / wasn't aware of the webpack implications here. Totally okay with it knowing that now.